mermithid/processors/misc/MultiGasComplexLineShape.py

```diff
@@ -0,0 +1,3748 @@
+        self.fixed_scatter_proportion = reader.read_param(params, 'fixed_scatter_proportion', True)
+        if self.fixed_scatter_proportion == True:
+            self.scatter_proportion = reader.read_param(params, 'gas_scatter_proportion', [])
+        self.partially_fixed_scatter_proportion = reader.read_param(params, 'partially_fixed_scatter_proportion', True)
+        if self.partially_fixed_scatter_proportion == True:
+            self.free_gases = reader.read_param(params, 'free_gases', ["H2", "He"])
+            self.fixed_gases = reader.read_param(params, 'fixed_gases', ["Ar", "Kr"])
+            self.gases = self.free_gases + self.fixed_gases
+            self.scatter_proportion_for_fixed_gases = reader.read_param(params, 'scatter_proportion_for_fixed_gases', [0.018, 0.039])
+        self.fixed_survival_probability = reader.read_param(params, 'fixed_survival_probability', True)
+        if self.fixed_survival_probability == True:
+            self.survival_prob = reader.read_param(params, 'survival_prob', 1)
+        self.use_radiation_loss = reader.read_param(params, 'use_radiation_loss', True)
+        self.sample_ins_resolution_errors = reader.read_param(params, 'sample_ins_res_errors', False)
+        # configure the resolution functions: gaussian_lorentzian_composite_resolution, elevated_gaussian, composite_gaussian, composite_gaussian_pedestal_factor, and simulated_resolution_scaled
+        self.resolution_function = reader.read_param(params, 'resolution_function', '')
+        if self.resolution_function == 'gaussian_lorentzian_composite_resolution':
+            self.ratio_gamma_to_sigma = reader.read_param(params, 'ratio_gamma_to_sigma', 0.8)
+            self.gaussian_proportion = reader.read_param(params, 'gaussian_proportion', 0.8)
+        if self.resolution_function == 'elevated_gaussian':
+            self.ratio_gamma_to_sigma = reader.read_param(params, 'ratio_gamma_to_sigma', 0.8)
+        if self.resolution_function == 'composite_gaussian' or 'composite_gaussian_pedestal_factor':
+            self.A_array = reader.read_param(params, 'A_array', [0.076, 0.341, 0.381, 0.203])
+            self.sigma_array = reader.read_param(params, 'sigma_array', [5.01, 13.33, 15.40, 11.85])
+        if self.resolution_function == 'simulated_resolution_scaled':
+            self.fit_recon_eff = reader.read_param(params, 'fit_recon_eff', False)
+        if self.resolution_function == 'simulated_resolution_scaled_fit_scatter_peak_ratio' or 'gaussian_resolution_fit_scatter_peak_ratio':
+            self.fixed_parameter_names = reader.read_param(params, 'fixed_parameter_names', [])
+            self.fixed_parameter_values = reader.read_param(params, 'fixed_parameter_values', [])
+            #self.elevation_factor = reader.read_param(params, 'elevation_factor', 20)
```

Comment:
what's the elevation factor? and can it be deleted?

Reply:
It seems to be used in the elevated_gaussian function to determine how much Lorentzian-ness is included.

@casesyh , is the elevated_gaussian function used for something? Or can it be deleted?

Reply:
The elevated_gaussian function wasn't used for Phase II analysis but I don't think it's necessary to delete it. It shows that we tested the resolution model.

Reply:
@casesyh , Right now the initialization of self.elevation_factor is commented out, so the processor is not really set up to use the elevated_gaussian function. I think that either there should be an option to initialize the elevation_factor variable, or the function should be deleted.

If you wish to keep the various resolution function options that we didn't use, I think it would help if each resolution function had a comment at the top explaining what it is and what context(s) it's useful for. In a case like the elevated_gaussian, you could also write, "Tested for ..." and then explain the context in which it's tested.

In general, more comments explaining functions in the processor would make it easier to navigate.